cmake/SetupBoost.cmake

```diff
@@ -3,12 +3,16 @@
 
 find_package(Boost 1.60.0 REQUIRED COMPONENTS program_options)
 
-spectre_include_directories(${Boost_INCLUDE_DIR})
-list(APPEND SPECTRE_LIBRARIES ${Boost_LIBRARIES})
 message(STATUS "Boost include: ${Boost_INCLUDE_DIRS}")
 message(STATUS "Boost libraries: ${Boost_LIBRARIES}")
 
 file(APPEND
   "${CMAKE_BINARY_DIR}/LibraryVersions.txt"
   "Boost Version:  ${Boost_MAJOR_VERSION}.${Boost_MINOR_VERSION}.${Boost_SUBMINOR_VERSION}\n"
   )
+
+add_library(Boost INTERFACE IMPORTED)
+set_property(TARGET Boost
+  APPEND PROPERTY INTERFACE_LINK_LIBRARIES ${Boost_LIBRARIES})
+set_property(TARGET Boost PROPERTY
+  INTERFACE_INCLUDE_DIRECTORIES ${Boost_INCLUDE_DIR})
```

Comment:
Boost provides a cmake config by default since v1.70.0 or if built with `boost-cmake`. This is automatically loaded by `find_package` if not explicitly disabled (see https://cmake.org/cmake/help/v3.15/module/FindBoost.html#boost-cmake). We should either disable this with `set(Boost_NO_BOOST_CMAKE ON)` (I have to do this manually on my machine because I have a more recent version of boost installed) or require and use the cmake config.

Reply:
There's no way we can rely on CMake 3.15 and boost 1.70 being available, I'll set the disable flag